rationale/message_block_layout.md

```diff
@@ -46,4 +46,9 @@ The last piece of the puzzle is determining _which_ row the message is placed at
 
 ### Caveats
 
-The message placement rules described above conflict with the first rule that shares must be ordered by namespace ID, as shares between two messages that are not placed adjacent to each other do not have a natural namespace they belong to. This is resolved by having special transactions the block producer includes that specify a range a zero-padding shares for a given namespace ID (which must be between the namespace IDs of the surrounding real shares, inclusive).
+The message placement rules described above conflict with the first rule that shares must be ordered by namespace ID, as shares between two messages that are not placed adjacent to each other do not have a natural namespace they belong to. This is resolved with some rules:
+1. the namespace ID for message shares must be "even" (i.e. the lowest bit must be `0`), and
+1. the namespace ID for shares between messages must be exactly `1` more than the namespace ID of the preceding message (i.e. must be identical save for the lowest bit being `1` instead of `0`), and.
+1. the namespace ID for shares between requests with a reserved namespace ID and messages must be [`TAIL_TRANSACTION_PADDING_NAMESPACE_ID`](../specs/consensus.md#constants). Note that this only applies for shares _between_ requests with a reserved namespace ID and messages, i.e. if there are no messages in the block then this rule does not apply.
```

Comment:
Why do you need a special rule for reserved namespace IDs? Also, what is a "request" in the context of a block?

Reply:
> Also, what is a "request" in the context of a block?

I also found that word confusing:
https://github.com/lazyledger/lazyledger-specs/pull/21#discussion_r419653996

Maybe we should instead just say Tx or rather "shares with a reserved namespace ID" instead of requests?

Reply:
Well, I use "request" because transactions aren't the only things that touch state: https://github.com/lazyledger/lazyledger-specs/pull/89#discussion_r529003291

Reply:
> Why do you need a special rule for reserved namespace IDs?

Actually, I realized there's an issue here: if two messages both have the same namespace ID, then they can't both be included in a block under this scheme unless they are aligned to some nice power of 2. It might have been a better idea to simply have a special transaction type that pays for padding, and revert the changes made in this PR to target #86.

> Maybe we should instead just say Tx or rather "shares with a reserved namespace ID" instead of requests?

"shares with a reserved namespace ID" works here.

Reply:
Reverted in 6d9648a as per https://github.com/lazyledger/lazyledger-specs/pull/89#discussion_r529009826.